src/graph.py

```diff
@@ -1,82 +1,109 @@
 
+
 class GNode(object):
     """Create a GNode Constructor."""
 
     def __init__(self, value):
         """Create an instance of a GNode with the attribues of a value and a list of edges."""
         self.value = value
-        self.edges = []
+        self.edges = set()
+
+
+class Edge(object):
+    """Create an Edge Constructor."""
+
+    def __init__(self, begin, end):
+        """Create an instance of an edge."""
+        self.begin = begin
+        self.end = end
+
+    def __eq__(self, other):
+        """Allow comparison of two equal edges."""
+        return self.begin == other.begin and self.end == other.end
+
+    def __ne__(self, other):
+        """Allow comparison of two unequal edges."""
+        return self.begin != other.begin or self.end != other.end
 
 
 class Graph(object):
     """Create a Graph Constructor."""
 
     def __init__(self):
         """Create an instance of a Graph that has a list of nodes."""
-        self.list_of_nodes = []
+        self.set_of_nodes = set()
 
     def add_node(self, new_node):
-        """Adds a new node to the graph"""
-        self.list_of_nodes.append(new_node)
+        """Add a new node to the graph."""
+        self.is_node(new_node)
```

Comment:
This approach is not very user friendly. Since all you're doing with the GNode class is storing a pair of value and set of neighbors, why not just allow your user to input any existing, familiar data type as the node name/value? You are requiring your user to learn about, import, initialize and always use your GNode class. You could just as simple initialize your GNode class within the method.